edx_proctoring/tests/test_views.py

```diff
@@ -2847,3 +2854,106 @@ def test_no_access(self):
+        # Create an exam attempt
+        proctored_exam = self._create_proctored_exam()
+        create_exam_attempt(proctored_exam.id, self.user_to_retire.id)
+        attempt = ProctoredExamStudentAttempt.objects.filter(user_id=self.user_to_retire.id).first()
+
+        # Run the retirement command
+        deletion_url = reverse('edx_proctoring:user_retirement_api', kwargs={'user_id': self.user_to_retire.id})
+        response = self.client.post(deletion_url)
+        assert response.status_code == 204
+
+        retired_attempt = ProctoredExamStudentAttempt.objects.filter(user_id=self.user_to_retire.id).first()
+        assert retired_attempt.student_name != attempt.student_name
+        assert retired_attempt.last_poll_ipaddr != attempt.last_poll_ipaddr
+
+    def test_retire_user_exam_attempt_history(self):
+        """ Retiring a user should obfuscate PII for exam attempt history and return a 204 status """
+        # Create and archive an exam attempt so it appears in the history table
+        proctored_exam = self._create_proctored_exam()
+        create_exam_attempt(proctored_exam.id, self.user_to_retire.id)
+        attempt = ProctoredExamStudentAttempt.objects.filter(user_id=self.user_to_retire.id).first()
+        attempt.delete_exam_attempt()
+        attempt_history = ProctoredExamStudentAttemptHistory.objects.filter(user_id=self.user_to_retire.id).first()
+
+        # Run the retirement command
+        response = self.client.post(self.deletion_url)
+        assert response.status_code == 204
+
+        retired_attempt_history = ProctoredExamStudentAttemptHistory \
+            .objects.filter(user_id=self.user_to_retire.id).first()
+        assert retired_attempt_history.student_name != attempt_history.student_name
+        assert retired_attempt_history.last_poll_ipaddr != attempt_history.last_poll_ipaddr
+
+    def test_retire_user_allowances(self):
+        """ Retiring a user should delete their allowances and return a 204 """
+        proctored_exam = self._create_proctored_exam()
+        add_allowance_for_user(proctored_exam.id, self.user_to_retire.id, 'a_key', 30)
+        assert len(ProctoredExamStudentAllowance.objects.filter(user=self.user_to_retire.id)) == 1
+
+        # Run the retirement command
+        response = self.client.post(self.deletion_url)
+        assert response.status_code == 204
+
+        assert len(ProctoredExamStudentAllowance.objects.filter(user=self.user_to_retire.id)) == 0
+
+    def test_retire_user_allowances_history(self):
+        """ Retiring a user should delete their allowances and return a 204 """
+        proctored_exam = self._create_proctored_exam()
+        add_allowance_for_user(proctored_exam.id, self.user_to_retire.id, 'a_key', 30)
+        add_allowance_for_user(proctored_exam.id, self.user_to_retire.id, 'a_key', 60)
+        assert len(ProctoredExamStudentAllowanceHistory.objects.filter(user=self.user_to_retire.id)) == 1
```

Comment:
You can also use `.count()` on the query_set, rather than `len`in' it

Reply:
Thanks @matthugs!